tests/baselines/reference/thisConditionalOnMethodReturnOfGenericInstance.errors.txt

```diff
@@ -0,0 +1,24 @@
+thisConditionalOnMethodReturnOfGenericInstance.ts(6,15): error TS2577: Return type annotation circularly references itself.
+
+
+==== thisConditionalOnMethodReturnOfGenericInstance.ts (1 errors) ====
+    class A<T> {
+        unmeasurableUsage!: {[K in keyof T]-?: T[K]};
+    }
+    
+    class B<T> extends A<T> {
+        method(): string | (this extends C ? undefined : null) {
+                  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

```

Comment:
Per OP: I plan to back out the change making this test error, since that fix is somewhat unrelated to this, even if it did motivate it.